cogs/member_stats.py

```diff
@@ -1,21 +1,82 @@
 import discord
-from discord.ext import commands
+from discord.ext import tasks, commands
 from pathlib import Path
 from .validation import is_moderator, is_mod_commands_channel
 import json
+import os
+from os import path
+from datetime import datetime
+from dateutil import tz
 
 
 class MemberStats(commands.Cog):
     def __init__(self, bot):
         self.bot = bot
+        self.status = 0
+
+    def cog_unload(self):
+        self.m_stats.cancel()
+
+    @commands.command()
+    async def m_stats_start(self, ctx):
+        self.m_stats.start(ctx)
+        self.status = 1
+        await ctx.send(f'statistics recording started')
+
+    @commands.command()
+    async def m_stats_stop(self, ctx):
+        self.m_stats.cancel()
+        self.status = 0
+        await ctx.send(f'statistics recording stopped')
+
+    @commands.command()
+    async def m_stats_status(self, ctx):
+        if self.status == 1:
+            await ctx.send(f'statistics recording is running')
+        else:
+            await ctx.send(f'statistics recording NOT running')
 
     @commands.command()
+    async def m_stats_read(self, ctx):
+        if self.status == 1:
+            await ctx.send(f'first stop statistics recording ($m_stats_stop)')
+        else:
+            if path.exists('./server_specific/member_stats.csv'):
+                with open('./server_specific/member_stats.csv',
+                          'r') as stats_file:
+                    contents = stats_file.read()
+                    await ctx.send('note: recorded timezone is PST (UTC -8)')
+                    await ctx.send(f'{contents}')
+                    stats_file.close()
+            else:
+                await ctx.send(f'file wiped')
+
+    @commands.command()
+    async def m_stats_clear(self, ctx):
+        def check(m):
+            return (
+                not m.content.startswith('$m')
+            )
+        await ctx.send(f'wipe ALL statistics? y(yes) / n(no)')
+
+        while True:
+            yes_no_msg = await self.bot.wait_for('message',  check=check)
+            if yes_no_msg.content not in ['Y',  'y',  'N',  'n']:
+                await ctx.send('enter either \'Y\' or \'N\'.')
+                continue
+            if yes_no_msg.content in ['N',  'n']:
+                break
+            if yes_no_msg.content in ['Y',  'y']:
+                await ctx.send(f'statistics file cleared')
+                os.remove('./server_specific/member_stats.csv')
+                break
+
+    @tasks.loop(seconds=3600.0)
     async def m_stats(self, ctx):
         if not await is_moderator(ctx):
             return
         if not await is_mod_commands_channel(ctx):
             return
-
```

Comment:
I've generally put blanklines after the moderator validation checks - I think it gives better readability.